### ISSUE TITLE
[Bug]: The loaded data on the querynode is not row-based balanced

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#30890]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20240226-bea4c3a4-amd64
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):  pulsar  
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1.  deploy milvus with 3 queryNodes
2. test:
   a. create collection -> index -> load -> insert 3m-128d data into 3 partitions, 1m data in each partition -> flush
   b. concurrent tests: search + delete: `id > 3m` + scene_insert_partition: There are a total of 20 partitions. Each locust user inserts 200 items into any partition at a time. The pk of the inserted data increases globally starting from 3,000,000.
```
 'concurrent_params': {'concurrent_number': 100,
                       'during_time': '10h',
                       'interval': 120,
                       'spawn_rate': None},
 'concurrent_tasks': [{'type': 'search',
                       'weight': 4,
                       'params': {'nq': 100,
                                  'top_k': 100,
                                  'search_param': {'ef': 128},
                                  'timeout': 120}},
                      {'type': 'delete',
                       'weight': 1,
                       'params': {'expr': 'id '
                                          '> '
                                          '3000000',
                                  'timeout': 120}},
                      {'type': 'scene_insert_partition',
                       'weight': 4,
                       'params': {'ni': 200,
                                  'timeout': 120,
                                  'partition_num': 20,
                                  'start_id': 3000000,
                                  'with_flush': False,
                                  'random_id': True,
                                  'random_vector': True}}]}
```
3. queryNode loaded entity num and mem usage shows that loaded data is not row-based balanced
![image](https://github.com/milvus-io/milvus/assets/27288593/e7a2b861-79ff-41dc-b751-3c21c88e11df)
![image](https://github.com/milvus-io/milvus/assets/27288593/773d9bd3-4e20-42bf-abdd-9df602a6988e)

4. court result  of `utility.get_query_segment_info`: {'node-5': 1105674, 'node-7': 11779954, 'node-1': 976836}
[segment_info.txt](https://github.com/milvus-io/milvus/files/14430119/segment_info.txt)

### Milvus Log

- argo:  https://argo-workflows.zilliz.cc/archived-workflows/qa/539cc153-d6ae-4cf7-b223-d220d8744c76?nodeId=level-zero-stable-16d
- grafana: [metrics of level-zero-multi-par-16](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=P1809F7CD0C75ACF3&var-cluster=&var-namespace=qa-milvus&var-instance=level-zero-multi-par-16&var-collection=All&var-app_name=milvus&from=1709004404170&to=1709101844170)
- pods:
```
level-zero-multi-par-16-etcd-0                                    1/1     Running       0               26h     10.104.34.137   4am-node37   <none>           <none>
level-zero-multi-par-16-etcd-1                                    1/1     Running       0               26h     10.104.24.123   4am-node29   <none>           <none>
level-zero-multi-par-16-etcd-2                                    1/1     Running       0               26h     10.104.19.116   4am-node28   <none>           <none>
level-zero-multi-par-16-milvus-datanode-5dd8bd468b-99kcg          1/1     Running       0               26h     10.104.6.29     4am-node13   <none>           <none>
level-zero-multi-par-16-milvus-datanode-5dd8bd468b-k62qw          1/1     Running       0               26h     10.104.15.220   4am-node20   <none>           <none>
level-zero-multi-par-16-milvus-indexnode-7d465fb89-c7zgz          1/1     Running       0               26h     10.104.19.121   4am-node28   <none>           <none>
level-zero-multi-par-16-milvus-indexnode-7d465fb89-q6qsz          1/1     Running       0               26h     10.104.24.130   4am-node29   <none>           <none>
level-zero-multi-par-16-milvus-mixcoord-84887bc859-fjjd4          1/1     Running       0               26h     10.104.34.150   4am-node37   <none>           <none>
level-zero-multi-par-16-milvus-proxy-9bc958bb7-d4nvj              1/1     Running       0               26h     10.104.15.219   4am-node20   <none>           <none>
level-zero-multi-par-16-milvus-querynode-0-b5b4dc664-6rmxx        1/1     Running       0               26h     10.104.6.30     4am-node13   <none>           <none>
level-zero-multi-par-16-milvus-querynode-0-b5b4dc664-rmkdz        1/1     Running       0               26h     10.104.15.221   4am-node20   <none>           <none>
level-zero-multi-par-16-milvus-querynode-0-b5b4dc664-rpktr        1/1     Running       0               26h     10.104.34.151   4am-node37   <none>           <none>
level-zero-multi-par-16-minio-0                                   1/1     Running       0               26h     10.104.34.144   4am-node37   <none>           <none>
level-zero-multi-par-16-minio-1                                   1/1     Running       0               26h     10.104.19.117   4am-node28   <none>           <none>
level-zero-multi-par-16-minio-2                                   1/1     Running       0               26h     10.104.15.214   4am-node20   <none>           <none>
level-zero-multi-par-16-minio-3                                   1/1     Running       0               26h     10.104.24.128   4am-node29   <none>           <none>
level-zero-multi-par-16-pulsar-bookie-0                           1/1     Running       0               26h     10.104.34.146   4am-node37   <none>           <none>
level-zero-multi-par-16-pulsar-bookie-1                           1/1     Running       0               26h     10.104.19.118   4am-node28   <none>           <none>
level-zero-multi-par-16-pulsar-bookie-2                           1/1     Running       0               26h     10.104.24.127   4am-node29   <none>           <none>
level-zero-multi-par-16-pulsar-bookie-init-xvjkj                  0/1     Completed     0               26h     10.104.34.128   4am-node37   <none>           <none>
level-zero-multi-par-16-pulsar-broker-0                           1/1     Running       0               26h     10.104.24.122   4am-node29   <none>           <none>
level-zero-multi-par-16-pulsar-proxy-0                            1/1     Running       0               26h     10.104.34.129   4am-node37   <none>           <none>
level-zero-multi-par-16-pulsar-pulsar-init-t9qlh                  0/1     Completed     0               26h     10.104.34.130   4am-node37   <none>           <none>
level-zero-multi-par-16-pulsar-recovery-0                         1/1     Running       0               26h     10.104.24.119   4am-node29   <none>           <none>
level-zero-multi-par-16-pulsar-zookeeper-0                        1/1     Running       0               26h     10.104.34.139   4am-node37   <none>           <none>
level-zero-multi-par-16-pulsar-zookeeper-1                        1/1     Running       0               26h     10.104.15.216   4am-node20   <none>           <none>
level-zero-multi-par-16-pulsar-zookeeper-2                        1/1     Running       0               26h     10.104.19.120   4am-node28   <none>           <none>
```

**Comment by yanliang567** [comment]
@ThreadDao does the new inserted data balanced among the 20 partitions?
/assign @weiliu1031 
/unassign

**Comment by ThreadDao** [comment]
@weiliu1031 I need you to confirm again. There seems to be some problems with the monitoring of the following two instances.
1. [metrics of level-zero-insert-op-71-6335](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=P1809F7CD0C75ACF3&var-cluster=&var-namespace=qa-milvus&var-instance=level-zero-insert-op-71-6335&var-collection=All&var-app_name=milvus&from=1710142112546&to=1710209733819)
![image](https://github.com/milvus-io/milvus/assets/27288593/48c0e74c-1d19-4874-a7f2-4bcbde460832)
![image](https://github.com/milvus-io/milvus/assets/27288593/1133ae9a-5ae9-4ee4-a7b2-f0980f01bc87)

2. [metrics of level-zero-upsert-op-1-7800](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=P1809F7CD0C75ACF3&var-cluster=&var-namespace=qa-milvus&var-instance=level-zero-upsert-op-1-7800&var-collection=All&var-app_name=milvus&from=1710141827781&to=1710214949185)
![image](https://github.com/milvus-io/milvus/assets/27288593/b9cab52a-0baa-4bfb-8da6-ebb973342bc6)
![image](https://github.com/milvus-io/milvus/assets/27288593/391cfd01-93c9-4b86-ab56-02e2568ccbe8)

**Comment by weiliu1031** [comment]
please verify this

**Comment by XuanYang-cn** [comment]
Not l0 related
/unassign

**Comment by ThreadDao** [comment]
fixed `milvus-io-master-773c64e-20240314`
![image](https://github.com/milvus-io/milvus/assets/27288593/14d27061-eadc-4a12-a6ce-48e04607cea0)
![image](https://github.com/milvus-io/milvus/assets/27288593/8df1afb2-1d2c-473e-bc38-0ecdaf13a801)
